app/views/IsAgentManagingEstateView.scala.html

```diff
@@ -15,31 +15,35 @@
  *@
 
 @import models.Mode
+@import views.html.components.{ErrorSummary, SubmitButton, InputYesNo}
 
 @this(
     main_template: MainTemplate,
-    formHelper: FormWithCSRF
+    formHelper: FormWithCSRF,
+    errorSummary: ErrorSummary,
+    inputYesNo: InputYesNo,
+    submitButton: SubmitButton
 )
 
 @(form: Form[_], mode: Mode, utr: String)(implicit request: Request[_], messages: Messages)
 
 @main_template(
-    title = s"${errorPrefix(form)}${messages("isAgentManagingEstate.title")}"
+    title = s"${errorPrefix(form)}${messages("isAgentManagingEstate.title")}",
+    showBackLink = true
     ) {
 
     @formHelper(action = controllers.routes.IsAgentManagingEstateController.onSubmit(mode), 'autoComplete -> "off") {
 
-        @components.back_link()
+        @errorSummary(form.errors, isYesNo = true)
 
-        @components.error_summary(form.errors)
-
-        <span class="govuk-caption-xl">@messages("isAgentManagingEstate.subheading", utr)</span>
-
-        @components.input_yes_no(
+        @inputYesNo(
             field = form("value"),
-            label = messages("isAgentManagingEstate.heading")
+            label = messages("isAgentManagingEstate.heading"),
+            caption = Some(messages("isAgentManagingEstate.subheading", utr)),
+            legendClass = Some("govuk-heading-l"),
+            legendAsHeading = false
```

Comment:
Should this be a true? The legend should be the heading?

Reply:
This is a weird one because it's still setting the heading in the InputYesNo component because of the way the heading and caption work. The was I've done it is so the mark up ends up like this...
```
<legend>
<span class="govuk-caption-l">${caption.get}</span>
<h1>${messages(label)}</h1>
</legend>
```
rather than like this... with the span inside the h1
```
<legend>
<h1>
<span class="govuk-caption-l">${caption.get}</span>
${messages(label)}
</h1>
</legend>
```
Due to how the fieldset deals with html.content: https://github.com/hmrc/play-frontend-govuk/blob/c691f49d91c3dd5941bf30a7b4a968b515952228/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/govukFieldset.scala.html#L26 

I was wondering for ages why one of the tests wasn't passing and this is why, not sure if I want to go into `claim-a-trust` and change that, because I lifted the original component from there - since it's probably marked up like the second one. I don't particularly like that legendAsHeading is set to false as it's confusing but it's only in the one place, and only because of the use of the caption...

I was tempted to set it to false in the component and remove that line in the view but it just hides it rather than addresses it.

Reply:
I have pushed a commit that removes it completely from the view / ignores the view value and added logic so that it's always false if there's a caption (because of the html content), but still true otherwise.